lib/reek/source/source_code.rb

```diff
@@ -21,10 +21,10 @@ class SourceCode
       # code   - Ruby code as String
       # origin - 'STDIN', 'string' or a filepath as String
       # parser - the parser to use for generating AST's out of the given source
-      def initialize(code, origin, parser: Parser::Ruby22)
-        @source = code
-        @origin = origin
-        @parser = parser
+      def initialize(args)
+        @source = args[:code]
+        @origin = args[:origin]
+        @parser = args[:parser] || Parser::Ruby22
```

Comment:
That's actually a step backwards - using Hashes for just slurping in all arguments is an anti-pattern since it decreases the expressiveness of the method signature (or even obliterates it, like in this case). Back in the day using the "hash-for-all-the-things" was idiomatic ruby since ruby didn't have keywords arguments, which it now does. So if we want to improve this method we should rather switch all arguments to keyword arguments (and keep the default value for parser in the signature as well) and not the other way round ;)